src/main/java/jenkins/plugins/http_request/HttpRequest.java

```diff
@@ -176,16 +196,33 @@ public String getValidResponseContent() {
     }
 
     @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
-        defineDefaultConfigurations();
+    public boolean perform(AbstractBuild<?,?> build, Launcher launcher, BuildListener listener)
+    throws InterruptedException, IOException
+    {
+        ResponseContentSupplier responseContentSupplier = performHttpRequest(build, listener);
+
+        final PrintStream logger = listener.getLogger();
+        logResponseToFile(build.getWorkspace(), logger, responseContentSupplier);
+        return true;
+    }
 
+    public ResponseContentSupplier performHttpRequest(Run<?,?> run, TaskListener listener)
+    throws InterruptedException, IOException
+    {
         final PrintStream logger = listener.getLogger();
+        this.listener = listener;
         logger.println("HttpMode: " + httpMode);
 
 
-        final EnvVars envVars = build.getEnvironment(listener);
-        final List<NameValuePair> params = createParameters(build, logger, envVars);
-        String evaluatedUrl = evaluate(url, build.getBuildVariableResolver(), envVars);
+        final EnvVars envVars = run.getEnvironment(listener);
+        final List<NameValuePair> params = createParameters(run, logger, envVars);
+        String evaluatedUrl;
+        if (run instanceof AbstractBuild<?, ?>) {
+            final AbstractBuild<?, ?> build = (AbstractBuild<?, ?>) run;
+            evaluatedUrl = evaluate(url, build.getBuildVariableResolver(), envVars);
+        } else {
+            evaluatedUrl = url;
+        }
```

Comment:
Arguably clearer to make `evaluatedUrl` an argument of `performHttpRequest`, avoiding need for a cast since `evaluate` would be done by the calling `perform`.